pymc/sampling.py

```diff
@@ -1217,7 +1220,7 @@ def _prepare_iter_population(
     step,
     start: Sequence[PointType],
     parallelize: bool,
-    tune=None,
+    tune: int,
```

Comment:
Why remove the optional?

Reply:
1. The (only) call to `_prepare_iter_population` is in a context where tune is already `int`.
2. It's passed down into `_iter_population` which assumed `tune: int`.